src/Microsoft.Diagnostics.Monitoring/ClientEndpointInfoSource.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Diagnostics.NETCore.Client;
+
+namespace Microsoft.Diagnostics.Monitoring
+{
+    internal sealed class ClientEndpointInfoSource : IEndpointInfoSourceInternal
+    {
+        public Task<IEnumerable<IEndpointInfo>> GetEndpointInfoAsync(CancellationToken token)
+        {
+            List<IEndpointInfo> endpointInfos = new List<IEndpointInfo>();
+            foreach (int pid in DiagnosticsClient.GetPublishedProcesses())
+            {
+                // CONSIDER: Generate a "runtime instance identifier" based on the pipe name
+                // e.g. pid + disambiguator in GUID form.
```

Comment:
Could you use the `ProcessInfo` command here to get the GUID?

Reply:
I can for .NET 5 runtimes, but it'll either still be Empty for .NET 3.1 or I'll have to generate the Guid. I'm going to save this for a future change when I pull in more information for identifying individual processes. The issue with using any commands in this method is that it'll slow down the response to this method.

I'll probably have to kick off the command for each instance and time box the response. Then I'll have to come up with some way of stating that the information is incomplete (or that can be inferred by the fact that it is an empty Guid).